packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/users/_components/formUser/AclsSelect.component.tsx

```diff
@@ -0,0 +1,212 @@
+import { UserAcl } from '@/types/cloud/project/database/opensearch';
+import {
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@/components/ui/form';
+import { useAclsSelectForm } from './useAclsSelectForm.hook';
+import { Input } from '@/components/ui/input';
+import { Button } from '@/components/ui/button';
+import { ScrollArea } from '@/components/ui/scroll-area';
+import { Badge } from '@/components/ui/badge';
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select';
+import { USER_CONFIG } from './user.constants';
+import { Alert, AlertDescription } from '@/components/ui/alert';
+
+interface AclsSelectProps {
+  value: UserAcl[] | undefined;
+  onChange: (newAcls: UserAcl[] | undefined) => void;
+}
+
+const AclsSelect = React.forwardRef<HTMLInputElement, AclsSelectProps>(
+  ({ value, onChange }, ref) => {
+    const { t } = useTranslation(
+      'pci-databases-analytics/services/service/users',
+    );
+    const patternInputRef = useRef<HTMLInputElement>(null);
+    const addAclBtnRef = useRef<HTMLButtonElement>(null);
+    const scrollListRef = useRef<HTMLUListElement>(null);
+    const { form, schema } = useAclsSelectForm({
+      existingAcls: value,
+    });
+    type ValidationSchema = z.infer<typeof schema>;
+
+    // Scroll to the bottom of the list when a new acl is added
+    useEffect(() => {
+      const scrollContainer = scrollListRef.current;
+      if (scrollContainer) {
+        const { lastChild } = scrollContainer;
+        if (lastChild instanceof Element) {
+          lastChild.scrollIntoView({ behavior: 'smooth' });
+        } else {
+          scrollContainer.scrollTop = scrollContainer.scrollHeight;
+        }
+      }
+    }, [value, scrollListRef]);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Enter') {
+        addAclBtnRef.current?.click();
+        patternInputRef.current?.focus();
+        event.preventDefault();
+      }
+    };
+    const handleAddAcl: SubmitHandler<ValidationSchema> = (formValues) => {
+      onChange([
+        ...value,
+        {
+          pattern: formValues.pattern,
+          permission: formValues.permission,
+        },
+      ]);
+      form.reset();
+      patternInputRef.current?.focus();
+    };
+    const handleRemoveAcl = (index: number) => {
+      const updatedAcls = [...value];
+      updatedAcls.splice(index, 1);
+      onChange(updatedAcls);
+    };
+
+    const errors = useMemo(() => {
+      const messages: string[] = [];
+      const formErrors = form.formState.errors;
+      Object.keys(formErrors).forEach((key) => {
+        const validKey = key as keyof typeof formErrors;
+        const formError = formErrors[validKey];
+        if (formError?.message) {
+          messages.push(formError.message);
+        }
+      });
+
+      return messages;
+    }, [form.formState.errors]);
```

Comment:
IMO, type cast is pretty weak because if `key` is not part of `formErrors` it could raise a silent error .
BTW, it could be simplified using filter + map to prevent from `push` mutability :)